File: Neetcode 150/Java/Array & Hashing/Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> compMap = new HashMap<>();
        int ans[] = {-1,-1};
        for(int i=0; i<nums.length; i++){
            if(compMap.containsKey(nums[i])){
                ans[0] = compMap.get(nums[i]);
                ans[1] = i;
                break;
            }
            int complement = target-nums[i];
            compMap.put(complement, i);
        }
        return ans;
    }
}

File: Neetcode 150/C++/Array & Hashing/Two Sum.cpp
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> mp;
        int n = nums.size();
        for (int i = 0; i < n; i++)
        {
            int remain = target - nums[i];
            if (mp.find(remain) != mp.end())
            {
                return {mp[remain], i};
            }
            mp.insert({nums[i], i});
        }
        return {};
    }
};

// T.c ==> O(n)
// S.c ==> O(n)
